packaging/libaec-config.cmake

```diff
@@ -0,0 +1,74 @@
+# libaec-config.cmake
+# -------------------
+#
+# Finds the AEC library, specify the starting search path in libaec_ROOT.
+#
+# Static vs. shared
+# -----------------
+# To make use of the static library instead of the shared one, one
+# needs to set the variable libaec_USE_STATIC_LIBS to ON before
+# calling find_package.
+# Example:
+#   set(libaec_USE_STATIC_LIBS ON)
+#   find_package(libaec CONFIG)
+#
+# This will define the following variables:
+#
+#   libaec_FOUND   - True if the system has the AEC library.
+#   libaec_VERSION - The version of the AEC library which was found.
+#   SZIP_FOUND     - True if the system has the SZIP library.
+#   SZIP_VERSION   - The version of the SZIP library which was found.
+#   SZIP_LIBRARIES - All the required libraries to make use of SZIP.
+#
+# and the following imported targets:
+#
+#   libaec::aec-shared  - The shared AEC library target (if it was built).
+#   libaec::sz-shared   - The shared SZIP compatible version of the AEC library
+#                         (if it was built).
+#   libaec::aec-static  - The static AEC library target (if it was built).
+#   libaec::sz-static   - The static SZIP compatible version of the AEC library
+#                         (if it was built).
+#   libaec::aec         - The (shared or static) AEC library target (according
+#                         to the value of libaec_USE_STATIC_LIBS).
+#   libaec::sz          - The (shared or static) SZIP compatible version of the
+#                         AEC library (according to the value of
+#                         libaec_USE_STATIC_LIBS).
+
+include(${CMAKE_CURRENT_LIST_DIR}/libaec-targets.cmake)
+
+# Support legacy variables
+find_path(SZIP_INCLUDE_DIR NAMES szlib.h DOC "SZIP include directory")
```

Comment:
To avoid potential inconsistencies if a user installed libaec at different prefixes simultaneously, it might make sense to extract that information from the import targets instead of trying to get it independently.
Untested! Might require some additional re-ordering
``` suggestion
get_target_property(SZIP_INCLUDE_DIR libaec::sz INTERFACE_INCLUDE_DIRECTORIES)
if (SZIP_INCLUDE_DIR)
  # This might be a list. The first item is the directory with the "main" header.
  list(GET SZIP_INCLUDE_DIR 0 SZIP_INCLUDE_DIR)
endif ()
```

Reply:
I seem to recall that this won't work with ALIAS targets. If that is true, you might need to extract that from `libaec::sz-shared` or `libaec::sz-static` (depending on whether they exist).